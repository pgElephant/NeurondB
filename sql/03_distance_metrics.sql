-- Test distance functions
SELECT vector_l2_distance('[0.0, 0.0]'::vector, '[3.0, 4.0]'::vector) > 0 AS l2_positive;
SELECT vector_cosine_distance('[1.0, 0.0]'::vector, '[0.0, 1.0]'::vector) >= 0 AS cosine_valid;
SELECT vector_inner_product('[1.0, 2.0, 3.0]'::vector, '[4.0, 5.0, 6.0]'::vector) > 0 AS inner_product_positive;
SELECT vector_l1_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'::vector) > 0 AS l1_positive;
SELECT vector_hamming_distance('[1.0, 0.0, 1.0]'::vector, '[1.0, 1.0, 0.0]'::vector) >= 0 AS hamming_valid;
SELECT vector_chebyshev_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'::vector);
SELECT vector_minkowski_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'::vector, 3.0) > 0 AS minkowski_positive;
