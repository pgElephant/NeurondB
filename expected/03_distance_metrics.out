-- Test distance functions
SELECT vector_l2_distance('[0.0, 0.0]'::vector, '[3.0, 4.0]'::vector) > 0 AS l2_positive;
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_l2_distance('[0.0, 0.0]'::vector, '[3.0, 4.0]'...
                                                ^
SELECT vector_cosine_distance('[1.0, 0.0]'::vector, '[0.0, 1.0]'::vector) >= 0 AS cosine_valid;
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_cosine_distance('[1.0, 0.0]'::vector, '[0.0, 1...
                                                    ^
SELECT vector_inner_product('[1.0, 2.0, 3.0]'::vector, '[4.0, 5.0, 6.0]'::vector) > 0 AS inner_product_positive;
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_inner_product('[1.0, 2.0, 3.0]'::vector, '[4.0...
                                                       ^
SELECT vector_l1_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'::vector) > 0 AS l1_positive;
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_l1_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'...
                                                ^
SELECT vector_hamming_distance('[1.0, 0.0, 1.0]'::vector, '[1.0, 1.0, 0.0]'::vector) >= 0 AS hamming_valid;
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_hamming_distance('[1.0, 0.0, 1.0]'::vector, '[...
                                                          ^
SELECT vector_chebyshev_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'::vector);
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_chebyshev_distance('[1.0, 2.0]'::vector, '[4.0...
                                                       ^
SELECT vector_minkowski_distance('[1.0, 2.0]'::vector, '[4.0, 6.0]'::vector, 3.0) > 0 AS minkowski_positive;
ERROR:  type "vector" does not exist
LINE 1: SELECT vector_minkowski_distance('[1.0, 2.0]'::vector, '[4.0...
                                                       ^
